searx/plugins/ahmia_filter.py

```diff
@@ -18,15 +16,14 @@
 def get_ahmia_blacklist():
     global ahmia_blacklist
     if not ahmia_blacklist:
-        with open(join(join(searx_dir, "data"), "ahmia_blacklist.txt"), 'r') as f:
-            ahmia_blacklist = f.read().split()
+        ahmia_blacklist = ahmia_blacklist_loader()
     return ahmia_blacklist
 
 
 def not_blacklisted(result):
     if not result.get('is_onion'):
         return True
-    result_hash = md5(urlparse(result.get('url')).hostname.encode()).hexdigest()
+    result_hash = md5(result['parsed_url'].hostname.encode()).hexdigest()
```

Comment:
Are you sure that `parsed_url` exists every time, even in case of offline engines?

Reply:
You are right, there is no parsed_url / url for offline engines according to :
https://github.com/searx/searx/blob/db703a0283ee169381aeea97c678e666ae508348/searx/results.py#L202-L202

I suggest to replace the line 27 by
```
parsed_url = result.get('parsed_url')
if not result.get('is_onion') or not parsed_url:
```

Reply:
Code updated